### PR TITLE
Folia Support - Folialib

### PR DESCRIPTION
**Pull request by Yomamaeatstoes**
### Type

- [ x] Internal change (Doesn't affect end-user).
- [ ] External change (Does affect end-user).
- [ ] Wiki (Changes towards the [Wiki]).
- [ ] Other: __________ 

### Benafits of this PR

- Folialib Dedicated library for folia ( https://github.com/TechnicallyCoded/FoliaLib ) 

- Has been tested, without breaking anything.

### CODE REVIEW

**Comment by Andre601** [comment]
This Pull request is honestly redundant.

You already mentioned there being another one open, that also adds Folia support, so having yet another open is not very useful.
Also, just because the PR was not merged yet doesn't mean that it isn't considered or looked into. There is more than just adding a library, changing code and call it a day. There is discussions behind the scenes regarding the best aproaches here.

tl;dr: This PR is not really needed here, but I'll leave it up to the others to decide on wether this one gets closed or picked over the other one.

**Comment by Yomamaeatstoes** [comment]
> This Pull request is honestly redundant.
> 
> You already mentioned there being another one open, that also adds Folia support, so having yet another open is not very useful. Also, just because the PR was not merged yet doesn't mean that it isn't considered or looked into. There is more than just adding a library, changing code and call it a day. There is discussions behind the scenes regarding the best aproaches here.
> 
> tl;dr: This PR is not really needed here, but I'll leave it up to the others to decide on wether this one gets closed or picked over the other one.

Thanks for the heads up. I don't know if this would make a difference, but if requested it can be changed to use native schedulers without any libraries, and a Folia expansion with low, median, high tps of the top 3 regions and a average tps of the server.

**Comment by Yomamaeatstoes** [comment]
https://github.com/ProjectUnified/MineLib "Another affiliated library, which is on Maven Central."
